src/unstract/sdk/file_storage/fs_impl.py

```diff
@@ -0,0 +1,178 @@
+from typing import Union
+
+import fsspec
+from fsspec import AbstractFileSystem
+
+from unstract.sdk.exceptions import FileOperationError
+from unstract.sdk.file_storage.constants import Common, FileSeekPosition
+from unstract.sdk.file_storage.fs_interface import FileStorageInterface
+from unstract.sdk.file_storage.helper import FileStorageHelper
+
+
+class FileStorage(FileStorageInterface):
+    # This class integrates fsspec library for file operations
+
+    fs: fsspec  # fsspec file system handle
+
+    def __init__(self, provider, credentials: dict() = {}):
+        self.fs = FileStorageHelper.file_storage_init(
+            provider=provider, credentials=credentials
+        )
+
+    def open(
```

Comment:
@gaya3-zipstack I suggest making use of context managers instead to cleanly open and close files - it'll be a pain to keep track / explicitly close the file handles by the caller

Reply:
This logic has changed now.